### PR TITLE
Update CITATION.cff

### PR DESCRIPTION
**Pull request by IAlibay**
Checklist
* [ ] Added a ``news`` entry

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -28,9 +28,15 @@ authors:
...
+  family-names: Horton
+  orcid: 'https://orcid.org/0000-0001-8694-7200'
+- given-names: Matthew
+  family-names: Thompson
+  orcid: 'https://orcid.org/0000-0002-1460-3983'
 title: "The Open Free Energy library"
-version: 1.0
-date-released: 2024-04-19
+version: 1.1.0
+date-released: 2024-09-25
```
Comment (IAlibay) [inline CITATION.cff:39]: Do I need to update the doi?

Comment (mikemhenry) [reply CITATION.cff:39]: pretty sure I have the DOI setup to point to "latest" I will check!

Comment (mikemhenry) [reply CITATION.cff:39]: From zenodo:

Cite all versions? You can cite all versions by using the DOI [10.5281/zenodo.8344247](https://zenodo.org/doi/10.5281/zenodo.8344247). This DOI represents all versions, and will always resolve to the latest one. [Read more](https://zenodo.org/help/versioning).

Comment (mikemhenry) [reply CITATION.cff:39]: so no need to update the DOI

**Comment by IAlibay** [comment]
Of all the time for mamba to fail on us!

**Review by mikemhenry — APPROVED** [review]

**Pull request merged**